network/wsNetwork_test.go

```diff
@@ -3757,3 +3761,139 @@ func TestWebsocketNetworkTelemetryTCP(t *testing.T) {
 	t.Log("closed detailsA", string(pcdA))
 	t.Log("closed detailsB", string(pcdB))
 }
+
+type mockServer struct {
```

Comment:
The mock structs and methods are largely copied from the websocket testing suite since I needed to set custom headers that wsNetwork expects in order to maintain a connection.